crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -254,13 +255,30 @@ where
         .with_context(|| format!("Handling newly declared classes for block {next:?}"))?;
         let t_declare = t_declare.elapsed();
 
+        let t_signature = std::time::Instant::now();
+        let signature = sequencer
+            .signature(block_hash.into())
+            .await
+            .with_context(|| format!("Fetch signature for block {next:?} from sequencer"))?;
+        let t_signature = t_signature.elapsed();
+
+        // An extra sanity check for the signature API.
+        anyhow::ensure!(
+            block_hash == signature.signature_input.block_hash,
+            "Signature block hash mismatch, actual {:x}, expected {:x}",
+            signature.signature_input.block_hash.0,
+            block_hash.0,
+        );
```

Comment:
Shoud we make this `warn!` instead? An issue here will block sync for non-p2p nodes. If yes, then we should probably not store the signature either if this fails. Bit of a pita.

Reply:
But an issue here is not a signature mismatch, but a sign of serious feeder gateway malfunction (we have received the signature for a different hash than we've requested)?

This check is the equivalent of the check we're doing for state updates (that we've received the state update for the block hash we've requested).

Reply:
True but the state update is actually required for sync while this data isn't. I guess I'm just worried there is a bug.. anywhere and now this halts things.

But you're right, we should just leave this as is I think.

Reply:
> True but the state update is actually required for sync while this data isn't.

Depends on how you view sync. It's not required if you don't ever want to use P2P. Otherwise it results in a database that you won't be able to use for serving incoming P2P requests.